tests/installation/welcome.pm

```diff
@@ -18,14 +18,17 @@ use testapi;
 use utils 'ensure_fullscreen';
 
 sub run() {
-    while (1) {
-        my @welcome_tags = ('inst-welcome-confirm-self-update-server', 'scc-invalid-url');
-        if (get_var('BETA')) {
-            push @welcome_tags, 'inst-betawarning';
-        }
-        else {
-            push @welcome_tags, 'inst-welcome';
-        }
+    my $iterations;
+
+    my @welcome_tags = ('inst-welcome-confirm-self-update-server', 'scc-invalid-url');
+    if (get_var('BETA')) {
+        push @welcome_tags, 'inst-betawarning';
+    }
+    else {
+        push @welcome_tags, 'inst-welcome';
+    }
+
+    while ($iterations++ < scalar(@welcome_tags)) {
```

Comment:
Very good point, thanks. I've added explicit asserts to fail with relevant error.